sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -179,51 +175,25 @@ private PlanCoordinator buildPlanCoordinator() throws ConfigStoreException {
                 failureMonitor,
                 overrideRecoveryPlanManagers));
 
+        // If decommissioning nodes, set up decommission plan:
+        DecommissionPlanFactory decommissionPlanFactory =
+                new DecommissionPlanFactory(serviceSpec, stateStore, taskKiller);
+        Optional<Plan> decommissionPlan = decommissionPlanFactory.getPlan();
+        if (decommissionPlan.isPresent()) {
+            // Set things up for a decommission operation.
+            offerAccepter.addRecorder(new DecommissionRecorder(stateStore, decommissionPlanFactory.getResourceSteps()));
+            planManagers.add(DefaultPlanManager.createProceeding(decommissionPlan.get()));
+        }
+
         // Other custom plan managers
         planManagers.addAll(plans.stream()
                 .filter(plan -> !plan.isDeployPlan())
-                .map(DefaultPlanManager::new)
+                .map(DefaultPlanManager::createInterrupted)
                 .collect(Collectors.toList()));
 
         return new DefaultPlanCoordinator(planManagers);
     }
 
-    private static void killUnneededAndPendingOverrideTasks(
```

Comment:
I don't think we can get rid of this.

Reply:
Yep, the integration tests were not failing appropriately.  I've made them fail and have put this method mostly back to make them pass again.